### ISSUE TITLE
krusader does not run, and does not run well...

### ISSUE THREAD

**Issue by alpha-bit** [NixOS/nixpkgs#178445]
### Describe the bug

Was not able to run krusader through nix-env -i ... , in a debian host.

And in nixos with `nix-shell -p krusader', all the icons are missing.

### Steps To Reproduce

1. install it
2. try to run it.

### Expected behavior

Run on host debian.

Having icons on host nixos.

### Screenshots

On debian host:

14:15:57.715-warning qt.glx unknown@0 # qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
14:15:57.721-warning qt.glx unknown@0 # qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
14:15:57.725-fatal default unknown@0 # Could not initialize GLX
Aborted

**Comment by peterhoeg** [comment]
krusader in a KDE session on NixOS is perfectly fine - it appears though that it's impure and looking for something in the global environment and not finding those without various KDE components present.

**Comment by alpha-bit** [comment]
yes ::: surely missing dependencies when installed as stand alone.

O.o ... yet if may not stand alone ::: should be part of a bigger package, and not be able to be installed alone... o.O?... or include all dependencies it needs to run... :) ... as expected from all packages... right?

**Comment by peterhoeg** [comment]
> should be part of a bigger package, and not be able to be installed alone…

It doesn’t declare all its dependencies, which it should. Running KDE applications on !NixOS isn’t very well tested which is why this has slipped through the cracks. We have issues with other applications as well (amarok comes to mind).

**Comment by milahu** [comment]
strace shows access to

```
/run/current-system/sw/share/kservices5/
/run/current-system/sw/share/kservices5/emoticonstheme_kde.desktop
/run/current-system/sw/share/icons/
/run/current-system/sw/share/icons/breeze-dark/index.theme
/run/current-system/sw/lib/qt-5.15.5/plugins/iconengines/
/run/current-system/sw/share/mime/generic-icons
```

`/run/current-system/sw/share/icons/` is probably found via `XCURSOR_PATH`

```
echo $XCURSOR_PATH | tr : $'\n' 
/nix/store/3n7kzs4b6kd3qxishy646igkw21y4csm-plasma-workspace-5.25.4/share/icons
/home/user/.icons
/home/user/.local/share/icons
/home/user/.nix-profile/share/icons
/home/user/.nix-profile/share/pixmaps
/etc/profiles/per-user/user/share/icons
/etc/profiles/per-user/user/share/pixmaps
/nix/var/nix/profiles/default/share/icons
/nix/var/nix/profiles/default/share/pixmaps
/run/current-system/sw/share/icons
/run/current-system/sw/share/pixmaps
~/.icons
/var/empty/share/icons
/var/empty/share/pixmaps
/var/empty/X11R6/lib/X11/icons
```

the `breeze-dark` theme is probably chosen from `~/.config/kdedefaults/kdeglobals`

```
grep -r breeze-dark ~/.config/kde*
/home/user/.config/kdedefaults/kdeglobals:Theme=breeze-dark
```
